languages/python/oso/polar/polar.py

```diff
@@ -247,6 +255,45 @@ def get_class(self, name):
         """
         return self.host.get_class(name)
 
+    def get_allowed_resources(self, actor, action, cls) -> list:
+        """
+        Returns all the resources the actor is allowed to perform action on.
+
+        :param actor: The actor for whom to collect allowed resources.
+
+        :param action: The action that user wants to perform.
+
+        :param cls: The type of the resources.
+
+        :return: A list of the unique allowed resources.
+        """
+        # Data filtering.
+        resource = Variable("resource")
+        # Get registered class name somehow
+        class_name = self.host.cls_names[cls]
+        constraint = Expression(
+            "And", [Expression("Isa", [resource, Pattern(class_name, {})])]
+        )
+        results = list(
+            self.query_rule(
```

Comment:
Yeah one day, not for this first version. You run into the same problem that the validation queries had where you need to run the query from the language side so that external checks all work. We can do this later.